yellowbrick/features/jointplot.py

```diff
@@ -14,308 +14,386 @@
-                        Use either color to colorize the lines on a per class basis or
-                        colormap to color them on a continuous scale.
-        =============   ==================================================================
-
-    xy_plot: one of {'hist'}, default: 'hist'
-        The type of plot to render along the x and y axes
-        Currently, the choice is hist
-
-    xy_args: dict, default: None
-        Keyword arguments used for customizing the x and y plots:
-
-        ==============  =====================================================
-        Property        Description
-        --------------  -----------------------------------------------------
-        alpha           transparency
-        facecolor_x     background color of the x axis
-        facecolor_y     background color of the y axis
-        bins            used to set up the number of bins for the hist plot
-        histcolor_x     used to set the color for the histogram on the x axis
-        histcolor_y     used to set the color for the histogram on the y axis
-        ==============  =====================================================
-
-    ratio: float, default: 5
-        Ratio of joint axis size to the x and y axes height
-
-    space: float, default: 0.2
-        Space between the joint axis and the x and y axes
+    ax : matplotlib Axes, default: None
+        The axes to plot the figure on. If None is passed in the current axes will be
+        used (or generated if required). This is considered the base axes where the
+        the primary joint plot is drawn. It will be shifted and two additional axes
+        added above (xhax) and to the right (yhax) if hist=True.
+
+    columns : int, str, [int, int], [str, str], default: None
+        Determines what data is plotted in the joint plot and acts as a selection index
+        into the data passed to ``fit(X, y)``. This data therefore must be indexable by
+        the column type (e.g. an int for a numpy array or a string for a DataFrame).
+
+        If None is specified then either both X and y must be 1D vectors and they will
+        be plotted against each other or X must be a 2D array with only 2 columns. If a
+        single index is specified then the data is indexed as ``X[columns]`` and plotted
+        jointly with the target variable, y. If two indices are specfied then they are
+        both selected from X, additionally in this case, if y is specified, then it is
+        used to plot the color of points.
+
+        Note that these names are also used as the x and y axes labels if they aren't
+        specified in the joint_kws argument.
+
+    correlation : str, default: 'pearson'
+        The algorithm used to compute the relationship between the variables in the
+        joint plot, one of: 'pearson', 'covariance', 'spearman', 'kendalltau'.
+
+    kind : str in {'scatter', 'hex'}, default: 'scatter'
```

Comment:
Might be nice to add as a todo to the docs (maybe #687) to add images that show a few different versions of JointPlot (e.g. with hex, with a few different correlation measures, and with the feature-feature-target heatmap option)

Reply:
I've updated #728 with this since I think that the documentation is going to be replaced with a bit more detail. I'll also update #687.